src/jupyterlabcontroller/storage/kubernetes/lab.py

```diff
@@ -93,23 +98,117 @@ async def delete_namespace(self, name: str) -> None:
         """
         await self._namespace.delete(name, wait=True)
 
-    async def delete_pod(self, name: str, namespace: str) -> None:
+    async def delete_pod(self, names: LabObjectNames) -> None:
         """Delete a pod from Kubernetes with a grace period.
 
         Parameters
         ----------
-        name
-            Name of the pod.
-        namespace
-            Namespace of the pod.
+        names
+            Names of lab objects.
 
         Raises
         ------
         KubernetesError
             Raised if there is some failure in a Kubernetes API call.
         """
         grace_period = LAB_STOP_GRACE_PERIOD
-        await self._pod.delete(name, namespace, grace_period=grace_period)
+        namespace = names.namespace
+        await self._pod.delete(names.pod, namespace, grace_period=grace_period)
```

Comment:
This is just my OCD here, but if you're going to use a temporary variable for either of podname or namespace, please use it for both.